### ISSUE TITLE
No error when process does not have correct privledges.

### ISSUE THREAD

**Issue by jpittis** [yaa110/rust-iptables#4]
I was surprised that the library did not report an error when it did not have permission to use `iptables`. Instead, it failed silently. Would you be alright with accepting a patch to return an error when a process does not have permission to use `iptables`.

**Comment by jpittis** [comment]
A bit more context now that I understand the problem better:

The reason this is problematic is that for the `new_chain` method (and probably others), a `false` return value is ambiguous. `false` might mean that the program does not have the correct privileges, but `false` is also returned when an existing chain is created for a second time. This means it's impossible for a program to tell the difference between "that chain already exists" vs "you don't have the right privileges".

The easiest way to hack around this would be to do a list call on program boot to ensure it has the right privileges, and then assume it does for the rest of the run. That's not very clean though.

**Comment by pitkley** [comment]
@jpittis thanks for adding more context. I quickly looked into the status codes my local `iptables` binary returns, and if I try to execute it with insufficient privileges, the return code is `4`. If on the other hand I for example try to create a chain that already exists, the return code is `1`.

What I'm gathering from this is that rather than simply returning `output.status.success()`, which returns `true` for an exit-code of `0` and false otherwise, we should maybe check if the error code was `1`, which is our `false` case.

Looking [at the source](https://git.netfilter.org/iptables/tree/include/xtables.h?h=032dc4a18ab86173847b6016baf0819ccd7641c5#n408), this seems to be something that we could be able to rely on. Can you verify if the return-codes on your side match what I am seeing?

**Comment by jpittis** [comment]
Thanks for the amazingly speedy response @pitkley. I'll try it out on my system to confirm, and then try out putting together a patch.

**Comment by jpittis** [comment]
```
$ sudo iptables --list foobar; echo $?
iptables: No chain/target/match by that name.
1
$ iptables --list; echo $?
Fatal: can't open lock file /run/xtables.lock: Permission denied
4
```

Confirmed! Gonna try writing a patch and report back!

**Comment by yaa110** [comment]
Fixed in v0.4.1
